### ISSUE TITLE
[NBug] Input string was not in a correct format.

### ISSUE THREAD

**Issue by NN---** [gitextensions/gitextensions#7146]
## Current behaviour

Crash

## Expected behaviour

No crash, diff working.

## Steps to reproduce

Show diff for a file inside stash.

## Error Details
```
System.FormatException: Input string was not in a correct format.
   at System.Number.StringToNumber(String str, NumberStyles options, NumberBuffer& number, NumberFormatInfo info, Boolean parseDecimal)
   at System.Number.ParseInt32(String s, NumberStyles style, NumberFormatInfo info)
   at GitUI.Editor.Diff.DiffLineNumAnalyzer.Analyze(String diffContent)
   at GitUI.Editor.Diff.DiffViewerLineNumberControl.DisplayLineNumFor(String diff)
   at GitUI.Editor.FileViewerInternal.SetText(String text, Action openWithDifftool, Boolean isDiff)
   at GitUI.Editor.FileViewer.<>c__DisplayClass98_0.<ViewPatch>b__1()
   at GitUI.Editor.FileViewer.ShowOrDeferAsync(Int64 contentLength, Func`1 showFunc)
   at Microsoft.VisualStudio.Threading.JoinableTaskFactory.ExecuteJob[T](Func`1 asyncMethod, JoinableTask job)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.Threading.JoinableTask.CompleteOnCurrentThread()
   at GitUI.Editor.FileViewer.ViewPatch(String text, Action openWithDifftool)
   at GitUI.CommandsDialogs.RevisionDiffControl.<ShowSelectedFileDiffAsync>d__35.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at GitUI.CommandsDialogs.RevisionDiffControl.<DiffFiles_SelectedIndexChanged>d__36.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Microsoft.VisualStudio.Threading.JoinableTaskFactory.SingleExecuteProtector.TryExecute()
```

## Additional information
Press F3 on python file in diff in stash

## Environment
- Git Extensions 3.2.1.6628
- Build bc07663c3c893f1d896520e0bdcfbe35cfe4b729
- Git 2.23.0.windows.1
- Microsoft Windows NT 10.0.18362.0
- .NET Framework 4.8.4010.0
- DPI 96dpi (no scaling)

**Comment by gerhardol** [comment]
How to recreate? (I cannot)
Is this in 3.1 too? There were changes related to this (not this file though).

Not reported before.

**Comment by NN---** [comment]
3.2.1
Not sure whether I canreproduce but I can try

**Comment by NN---** [comment]
Did you try running external diff on file in stash?

**Comment by gerhardol** [comment]
> Did you try running external diff on file in stash?

Yes (I stashed a .py file change)
The diff does not have an expected format, it is likely special to your setup
The stack trace points to the internal file viewer, not a difftool

**Comment by NN---** [comment]
Tried reproducing but no luck :(

**Comment by gerhardol** [comment]
If this is caused in the same as #8634, it was fixed in #8661
Previous occurrences with explanation has been related to use of UTF-7 (that is removed in 3.5)
